src/journal/compress.c

```diff
@@ -667,13 +677,237 @@ int decompress_stream_lz4(int in, int out, uint64_t max_bytes) {
         return -EPROTONOSUPPORT;
 #endif
 }
+#if HAVE_ZSTD
+static int zstd_ret_to_errno(size_t ret) {
+        static_assert(ZSTD_VERSION_NUMBER >= 10301,
+                      "<zstd_errors.h> is stable since zstd 1.3.1");
+
+        ZSTD_ErrorCode error = ZSTD_getErrorCode(ret);
+
+        switch (error) {
+        case ZSTD_error_dstSize_tooSmall:
+                return -ENOBUFS;
+        case ZSTD_error_memory_allocation:
+                return -ENOMEM;
+        default:
+                return -EBADMSG;
+        }
+}
+#endif
+
+int compress_stream_zstd(int fdf, int fdt, uint64_t max_bytes) {
+#if HAVE_ZSTD
+        _cleanup_(ZSTD_freeCCtxp) ZSTD_CCtx *cctx = NULL;
+        _cleanup_free_ void *buffIn = NULL;
+        _cleanup_free_ void *buffOut = NULL;
+        size_t buffInSize, buffOutSize;
+        size_t z;
+        uint64_t left = max_bytes, in_bytes = 0;
+        uint64_t inputSize = 0;
+
+        assert(fdf >= 0);
+        assert(fdt >= 0);
+
+        /* Create the context and buffers */
+        buffInSize = ZSTD_CStreamInSize();
+        buffOutSize = ZSTD_CStreamOutSize();
+        buffIn = malloc(buffInSize);
+        buffOut = malloc(buffOutSize);
+        cctx = ZSTD_createCCtx();
+        if (!cctx || !buffOut || !buffIn)
+                return -ENOMEM;
+
+        if (inputSize) {
+                z = ZSTD_CCtx_setPledgedSrcSize(cctx, inputSize);
+                if (z)
+                        log_debug("Failed to enable Zstd Inputsize: %s",
```

Comment:
"Inputsize" is entirely new capitalization... maybe just write out proper English, i.e. "input size"

also, please say that the error is ignored in the msg, so that people who look at the debug logs won't be unnecessarily tripped up by this. i.e. "Failed to enable zsdt input size, ignoring: %s"

Reply:
(here and everywhere else, where errors are knowingly ignored)

Reply:
Me no english? thats unpossible!
Fixed up the issues, also consistently using uppercase ZSTD in the messages (like with XZ, LZ4)